core/imds/src/main/java/software/amazon/awssdk/imds/internal/DefaultEc2MetadataClient.java

```diff
@@ -149,6 +151,29 @@ public Ec2MetadataResponse get(String path) {
         throw sdkClientExceptionBuilder.build();
     }
 
+    private void handleUnsuccessfulResponse(int statusCode, Optional<AbortableInputStream> responseBody,
+                                            HttpExecuteResponse response, String path, boolean isTokenRequest) {
+        String responseContent = responseBody.map(BaseEc2MetadataClient::uncheckedInputStreamToUtf8)
+                                             .orElse("");
+
+        if (!responseContent.isEmpty()) {
+            log.debug(() -> String.format("%s request response body: %s",
+                                          isTokenRequest ? "Token" : "Metadata",
+                                          responseContent));
```

Comment:
I understand this is existing code, but we should remove this log. Customers can enable wirelogs if they want to see payload.